gomint-server/src/main/java/io/gomint/server/util/Precondition.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020, GoMint, BlackyPaw and geNAZt
+ *
+ * This code is licensed under the BSD license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+package io.gomint.server.util;
+
+import io.gomint.server.world.WorldAdapter;
+import io.gomint.world.World;
+
+import javax.annotation.Nullable;
+
+/**
+ * @author Janmm14
+ * @version 1.0
+ */
+public class Precondition {
+
+    private Precondition() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Throws {@linkplain UnsafeWorldAsyncAccessException} when given {@code world} is not null and its
+     * {@linkplain World#mainThread()} method returns false.
+     *
+     * @param world the world to enforce sync access to
+     */
+    public static void safeWorldAccess(@Nullable World world) {
+        if (world != null && !world.mainThread()) {
+            throw new UnsafeWorldAsyncAccessException(world.folder());
+        }
+    }
+
+    /**
+     * Throws {@linkplain UnsafeWorldAsyncAccessException} when given {@code world} is not null and its
+     * {@linkplain WorldAdapter#hardMainThreadCheck()} method returns false.
+     *
+     * @param world the world to enforce sync access to
+     */
+    public static void safeWorldAccessHard(@Nullable WorldAdapter world) {
```

Comment:
Can we rename this? This is only used to allow the I/O thread. This could be a parameter "allowForIOThread" in safeWorldAccess and the mainThread() check

Reply:
will do

Reply:
done